exporter/awsemfexporter/datapoint.go

```diff
@@ -156,6 +164,79 @@ func (dps histogramDataPointSlice) CalculateDeltaDatapoints(i int, instrumentati
 	}}, true
 }
 
+// CalculateDeltaDatapoints retrieves the ExponentialHistogramDataPoint at the given index.
+func (dps exponentialHistogramDataPointSlice) CalculateDeltaDatapoints(i int, instrumentationScopeName string, detailedMetrics bool) ([]dataPoint, bool) {
+	metric := dps.ExponentialHistogramDataPointSlice.At(i)
+	labels := createLabels(metric.Attributes(), instrumentationScopeName)
+	timestamp := unixNanoToMilliseconds(metric.Timestamp())
+
+	scale := metric.Scale()
+	base := math.Pow(2, math.Pow(2, float64(-scale)))
+	datapoints := []dataPoint{}
+	arrayValues := []float64{}
+	arrayCountes := []float64{}
+
+	// Set mid-point of positive buckets in values/counts array.
+	positiveBuckets := metric.Positive()
+	positiveOffset := positiveBuckets.Offset()
+	positiveBucketCounts := positiveBuckets.BucketCounts()
+	for i := 0; i < positiveBucketCounts.Len(); i++ {
+		index := i + int(positiveOffset)
+		bucketBegin := math.Pow(base, float64(index))
+		bucketEnd := math.Pow(base, float64(index+1))
```

Comment:
Does this repeatedly calculate the same value for successive buckets?  Can that be avoided?  One bucket ends where the next begins.

Reply:
Yes, it will be optimized to calculate bucket boundary only once.